### PR TITLE
Fix/analytics

### PR DESCRIPTION
**Pull request by rcapeto**
#### What is the purpose of this pull request?

Fix {{ecommerce}} variable firing, pass null to {{ecommerce}} before pass values again, like google recommend.

#### What problem is this solving?

Check if has orderId in orders table in master data.

#### How should this be manually tested?

[Workspace](https://fixgtmv2--brmotorolanew.myvtex.com/)

#### Screenshots or example usage

-- no need

#### Types of changes

* [x] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by rcapeto** [comment]
> Hi! I'm VTEX IO CI/CD Bot and I'll be helping you to publish your app! 🤖
> 
> Please select which version do you want to release:
> 
> * [x]  Patch (backwards-compatible bug fixes)
> * [ ]  Minor (backwards-compatible functionality)
> * [ ]  Major (incompatible API changes)
> 
> And then you just need to merge your PR when you are ready! There is **no need** to create a release commit/tag.
> 
> * [ ]  No thanks, I would rather do it manually 😞

**Review by leeoxleeo — APPROVED** [review]

**Review by leeoxleeo — APPROVED** [review]

**Review by victorhmp — CHANGES_REQUESTED** [review]
Please update the CHANGELOG

File: react/modules/updateEcommerce.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+   (eventIndex >= 0) && window.dataLayer.splice(eventIndex, 1);
+
+   if(Object.keys(data).includes('ecommerce')) {
+      const ecommerceNullIndex = window.dataLayer.findIndex(gtmEvent => gtmEvent.ecommerce === null);
+      (ecommerceNullIndex >= 0) && window.dataLayer.splice(ecommerceNullIndex, 1);
+      push({ ecommerce: null });
+   }
+   
+   push(data);
+}
```
Comment (victorhmp) [inline react/modules/updateEcommerce.ts:17]: Please lint this file. We don't use `;`s

File: react/modules/updateEcommerce.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import push from './push';
+
+export default function updateEcommerce(
+   eventName: string, 
+   data: Record<string, unknown>
+) {
+   const eventIndex = window.dataLayer.findIndex(gtmEvent => gtmEvent.event === eventName);
+   (eventIndex >= 0) && window.dataLayer.splice(eventIndex, 1);
```
Comment (victorhmp) [inline react/modules/updateEcommerce.ts:8]: Add and actual `if` here to make it more explicit

File: react/modules/updateEcommerce.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+   if(Object.keys(data).includes('ecommerce')) {
+      const ecommerceNullIndex = window.dataLayer.findIndex(gtmEvent => gtmEvent.ecommerce === null);
+      (ecommerceNullIndex >= 0) && window.dataLayer.splice(ecommerceNullIndex, 1);
```
Comment (victorhmp) [inline react/modules/updateEcommerce.ts:12]: Same here

File: react/modules/updateEcommerce.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      push({ ecommerce: null });
```
Comment (victorhmp) [inline react/modules/updateEcommerce.ts:13]:
```suggestion

      push({ ecommerce: null });
```

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Comment by victorhmp** [comment]
Forgot to comment on it, but the tests are also failing, please take a look at them too :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+## [3.3.1] - 2022-02-11
+
```
Comment (victorhmp) [inline CHANGELOG.md:11]: Remove the version number and date

```suggestion
## [Unreleased]

```

**Review by victorhmp — APPROVED** [review]
LGTM, but please apply the suggestion to fix the CHANGELOG

**Pull request merged**